charuco_detector:
  ros__parameters:
    charuco_parameters:
      squaresSidesSizeM: 0.0200
      markersSidesSizeM: 0.0150
      numberOfBitsForMarkersSides: 4
      numberOfMarkers: 70
      numberOfSquaresInX: 10
      numberOfSquaresInY: 14
      dictionaryId: 3
    function_parameters:
      use_median_blur: false
      median_blur_k_size: 6
      use_dynamic_range: false
      use_bilateral_filter: false
      bilateral_filter_pixel_neighborhood: 5
      bilateral_filter_sigma_color: 100.0
      bilateral_filter_sigma_space: 100.0
      bilateral_filter_border_type: 4
      use_clahe: true
      clahe_clip_limit: 1.0
      clahe_size_x: 2
      clahe_size_y: 2
      use_adaptive_threshold: false
      adaptive_threshold_max_value: 255.0
      adaptive_threshold_method: 1
      adaptive_threshold_type: 0
      adaptive_threshold_block_size: 65
      adaptive_threshold_constant_offset_from_mean: 0.0
    tf_parameters:
      use_static_tf_broadcaster: true
      tf_broadcaster_republish_rate: 10.0
      sensor_frame_override: calib_camera
      charuco_tf_frame: calib_charuco

    subscribers:
      image_topic:
        topic: /camera/color/image_raw
        queue_size: 1
      camera_info:
        topic: /camera/color/camera_info
        queue_size: 1

    publishers:
      image_results_publisher:
        topic: ""
        queue_size: 1
        latch: false
      charuco_pose_publisher:
        topic: ""
        queue_size: 1
        latch: false